pkg/math/voteweighted/voteweighted.go

```diff
@@ -194,3 +209,63 @@ func ComputeMedian(priceInfo PriceInfo) *big.Int {
 
 	return nil
 }
+
+func ConstrainedSWMedian(
+	logger log.Logger,
+	validatorStore ValidatorStore,
+	threshold math.LegacyDec,
+	oracleKeeper OracleKeeper,
+	thresholdDetermination ThresholdDetermination,
+) aggregator.AggregateFnFromContext[string, map[slinkytypes.CurrencyPair]*big.Int] {
+	return func(ctx sdk.Context) aggregator.AggregateFn[string, map[slinkytypes.CurrencyPair]*big.Int] {
+		// providers is a map of consensus-address to map of CurrencyPair to Price
+		// it contains the mappings of price points for each validator
+		return func(providers aggregator.AggregatedProviderData[string, map[slinkytypes.CurrencyPair]*big.Int]) map[slinkytypes.CurrencyPair]*big.Int {
+			// Calculate map of CurrencyPair to PriceInfo
+			priceInfo := PriceInfoFromProviders(ctx, logger, validatorStore, providers)
+			// Iterate through all prices and compute the median price for each asset.
+			prices := make(map[slinkytypes.CurrencyPair]*big.Int)
+			totalBondedTokens, err := validatorStore.TotalBondedTokens(ctx)
+			if err != nil {
+				// This should never error.
+				panic(err)
+			}
+			for currencyPair, info := range priceInfo {
+				// The total voting power % that submitted a price update for the given currency pair must be
+				// greater than the threshold to be included in the final oracle price.
+				// The thresholdDetermination function is used to alter the considered weight of votes.
+				quote, err := oracleKeeper.GetPriceForCurrencyPair(ctx, currencyPair)
+				if err != nil {
+					logger.Error(
+						"found currency pair in votes which doesn't exist in module state",
+						"currency pair", currencyPair.String(),
+						"error", err,
+					)
+					continue
+				}
+				currentPrice := quote.Price.BigInt()
+				// newPrice is the calculated stake-weighted median
+				newPrice := ComputeMedian(info)
```

Comment:
In the event that we don't have enough stake  to compute a stake weighted median (i.e info.TotalWeight < 2/3 stake weight report vote-extensions with prices), should we even compute a stake weighted median / perform the threshold calculation?